packages/scripts/src/commands/app-data/convert/index.ts

```diff
@@ -84,71 +90,75 @@ export class AppDataConverter {
-      description: string;
-      fn: (context: any) => Promise<any>;
-    }[] = [
-      {
-        name: "xlsx_read",
-        description: "Load list of all input xlsx workbooks from folder",
-        fn: async () => {
-          const filterFn = (relativePath: string) => relativePath.endsWith(".xlsx");
-          return generateFolderFlatMap(inputFolder, { filterFn });
-        },
-      },
-      {
-        name: "xlsx_convert",
-        description: "Convert all xlsx workbooks to json arrays",
-        fn: async (list: { [key: string]: IContentsEntry }) => {
-          const xlsxConverter = new XLSXWorkbookProcessor(converterPaths);
-          return xlsxConverter.process(Object.values(list));
-        },
-      },
-      {
-        name: "jsons_clean",
-        description: "Flatten json arrays to single list of all sheets, filter and sort",
-        fn: async (data: FlowTypes.FlowTypeWithData[][]) => this.cleanFlowOutputs(data),
-      },
-      {
-        name: "flows_process",
-        description: "Apply specific parsers for flowtypes",
-        fn: async (data: FlowTypes.FlowTypeWithData[]) => {
-          const processor = new FlowParserProcessor(converterPaths);
-          return processor.process(data);
-        },
-      },
-    ];
-    let input: any;
-    let output: any;
-    clearLogs();
-    for (const step of processPipeline) {
-      output = await step.fn(input);
-      this.logger.debug({ step: step.name, output });
-      input = output;
+    // Process each input folder, converting xlsx to json
+    for (const inputFolder of inputFolders) {
+      const folderOutputsHashmap: Record<string, FlowTypes.FlowTypeWithData> = {};
+      converterPaths.SHEETS_INPUT_FOLDER = inputFolder;
+      const list = generateFolderFlatMap(inputFolder, { filterFn });
+      const xlsxConverter = new XLSXWorkbookProcessor(converterPaths);
+      xlsxConverter.logger = this.logger;
+      const data = await xlsxConverter.process(Object.values(list));
+      const outputs = this.cleanFlowOutputs(data);
+      // merge outputs across folders, with latter processed overriding any flows with same type and name
+      // keep local folder list to still track case where duplicate flows appear in same folder
+      for (const output of outputs) {
+        const hashName = `${output.flow_type}||${output.flow_name}`;
+        if (folderOutputsHashmap[hashName]) {
+          this.logDuplicateFlowError(folderOutputsHashmap[hashName], output);
+        }
+        folderOutputsHashmap[hashName] = output;
+        if (combinedOutputsHashmap[hashName]) {
+          this.logger.debug({
+            message: "flow_override",
+            details: { flow_type: output.flow_type, flow_name: output.flow_name, inputFolder },
+          });
+        }
+        combinedOutputsHashmap[hashName] = output;
+      }
+      this.logger.debug({ step: inputFolder, outputs });
     }
-    const result = output as IParsedWorkbookData;
-
+    // Process jsons as flows
```

Comment:
It could be worth making these comments even more prominent – I quite like the way that the previous `processPipeline` syntax broke down the pipeline into clear steps. Although it may be that good documentation would be better than comments: I have added #1520 to my list of high priorities

Reply:
Yes I had to remove the step-by-step mode as we needed to more deeply nest the processing (inner loop to extract sheets within a given source and outer loop to merge across sources). I'll add a few more comments though for clarity